gax-java/gax/src/main/java/com/google/api/gax/rpc/Watchdog.java

```diff
@@ -220,6 +220,16 @@ public void onStartImpl(StreamController controller) {
             public void disableAutoInboundFlowControl() {
               Preconditions.checkState(
                   !hasStarted, "Can't disable automatic flow control after the stream has started");
+
+              synchronized (lock) {
```

Comment:
I dont think you need to lock here. While we are in !hasStarted mode, the only relevant thread is the caller's